primitives/core/src/traits.rs

```diff
@@ -46,27 +48,28 @@ pub enum BareCryptoStoreError {
 }
 
 /// Something that generates, stores and provides access to keys.
+#[async_trait]
 pub trait BareCryptoStore: Send + Sync {
 	/// Returns all sr25519 public keys for the given key type.
-	fn sr25519_public_keys(&self, id: KeyTypeId) -> Vec<sr25519::Public>;
+	async fn sr25519_public_keys(&self, id: KeyTypeId) -> Vec<sr25519::Public>;
```

Comment:
Note that all methods here will most likely end up returning a `Result` so that we can handle potential remote signer issues (unless we want them to fall back to some defaults).